coordinator/internal/transitengineapi/transitengineapi_test.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2024 Edgeless Systems GmbH
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package transitengine
+
+import (
+	"bytes"
+	"encoding/json"
+	"log/slog"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/edgelesssys/contrast/coordinator/internal/authority"
+	"github.com/edgelesssys/contrast/coordinator/internal/seedengine"
+	"github.com/edgelesssys/contrast/internal/constants"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTransitAPICyclic(t *testing.T) {
+	type requestSpecs struct {
+		name           string
+		Plaintext      string `json:"plaintext"`
+		AssociatedData string `json:"associated_data,omitempty"`
+		Version        int    `json:"key_version"`
+		expStatus      int
+	}
+	mocks := map[string]requestSpecs{
+		"positive": {
+			Plaintext:      "vpIhKQhFuGwLv5B/XLYr960uZQ==",
+			name:           "autounseal",
+			AssociatedData: "Xv0nZLWkSan+vdWrH2LrGP8TU/Qg1+ZX7vldWMbxKTk=",
+			Version:        2,
+			expStatus:      200,
+		},
+		"special char name": {
+			Plaintext:      "lT3rQGMlxq680DdSKfIYYcfyCfMnP9ikxaO5b0mGRKRl4qNL3W9xkSW3QmaMwozCRfNMZhhDCbYokn6KEiGotlVInKt66QjBgXR2Nk9hIcez0LYt8W5pxD0lwTxC",
+			name:           "thi$$hoU_ld+*work",
+			AssociatedData: "HribV+LFZspJpAauFf643A1HKbj1VlQWVhAKFDJqdZg=",
+			Version:        2000,
+			expStatus:      200,
+		},
+		"f empty name": {
+			Plaintext:      "vpIh",
+			name:           "",
+			AssociatedData: "vpIh",
+			expStatus:      404,
+		},
+		"f name with forsl": {
+			Plaintext:      "vpIh",
+			name:           "wrong/url",
+			AssociatedData: "vpIh",
+			expStatus:      404,
+		},
+		"f bad request nob64plain": {
+			Plaintext:      "thi$$hoU_ld+*notwork",
+			name:           "autounseal",
+			AssociatedData: "vpIh",
+			expStatus:      400,
+		},
+		"f bad request nob64AD": {
```

Comment:
`nob64AD` the AD stands for additional data, right? I think we should spell this out and more generally be more verbose i.e. name this "failed bad request no base64 additional data"